File: Leetcode/853. Car Fleet.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        int n = position.size();
        if (n == 0) return 0;
        
        vector<pair<int, double>> v;  
        for (int i = 0; i < n; i++) {
            double time = (double)(target - position[i]) / speed[i];  
            v.push_back({position[i], time});
        }
        
        sort(v.rbegin(), v.rend());  

        int res = 0;
        double prevTime = 0;
        for (auto& car : v) {
            double currentTime = car.second;
            if (currentTime > prevTime) {  
                res++;  
                prevTime = currentTime;  
            }
        }
        
        return res;
    }
};

File: Leetcode/853. Car Fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        pairs=[[p,s] for p,s in zip(position,speed)]
        pairs.sort(reverse=True)
        st=[]
        for p,s in pairs:
            st.append((target-p)/s)
            if len(st)>=2 and st[-1]<=st[-2]:
                st.pop()
        return len(st)
